From 4712013e9865054553e337d3f269124de9236a6e Mon Sep 17 00:00:00 2001
From: Alexander Lobakin <alobakin@pm.me>
Date: Sun, 9 Oct 2022 18:55:21 +0200
Subject: [PATCH 12/14] i2c: i2c-nct6775: fix -Wimplicit-fallthrough

Depending on a compiler, `case I2C_SMBUS_BYTE_DATA` can trigger
-Wimplicit-fallthrough / unannotated fallthrough as it doesn't
mention it anyhow.
Add a `fallthrough` attribute to silence this.

Fixes: 480fefb8f23ab ("i2c: busses: Add SMBus capability to work with OpenRGB driver control")
Signed-off-by: Alexander Lobakin <alobakin@pm.me>
---
 drivers/i2c/busses/i2c-nct6775.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i2c/busses/i2c-nct6775.c b/drivers/i2c/busses/i2c-nct6775.c
index 0462f0952..e919d1e10 100644
--- a/drivers/i2c/busses/i2c-nct6775.c
+++ b/drivers/i2c/busses/i2c-nct6775.c
@@ -219,6 +219,7 @@ static s32 nct6775_access(struct i2c_adapter * adap, u16 addr,
 			break;
 		case I2C_SMBUS_BYTE_DATA:
 			tmp_data.byte = data->byte;
+			fallthrough;
 		case I2C_SMBUS_BYTE:
 			outb_p((addr << 1) | read_write,
 			       SMBHSTADD);
-- 
2.40.1.445.gf85cd430b1

